app/src/Registry/App/API.purs

```
@@ -1139,25 +1144,31 @@ pursPublishMethod :: PursPublishMethod
 pursPublishMethod = LegacyPursPublish
 
 fetchPackageSource
-  :: { tmpDir :: FilePath, ref :: String, location :: Location }
-  -> RegistryM { packageDirectory :: FilePath, publishedTime :: DateTime }
+  :: forall m r
+   . MonadRegistry m
+  => MonadAsk (GitHubEnv r) m
+  => { tmpDir :: FilePath, ref :: String, location :: Location }
+  -> m { packageDirectory :: FilePath, publishedTime :: DateTime }
 fetchPackageSource { tmpDir, ref, location } = case location of
   Git _ -> do
     -- TODO: Support non-GitHub packages. Remember subdir when doing so. (See #15)
-    throwWithComment "Packages are only allowed to come from GitHub for now. See #15"
+    Notify.die "Packages are only allowed to come from GitHub for now. See #15"
 
   GitHub { owner, repo, subdir } -> do
     -- TODO: Support subdir. In the meantime, we verify subdir is not present. (See #16)
-    when (isJust subdir) $ throwWithComment "`subdir` is not supported for now. See #16"
+    when (isJust subdir) $ Notify.die "`subdir` is not supported for now. See #16"
 
     case pursPublishMethod of
-      LegacyPursPublish -> liftAff do
-        log $ "Cloning repo at tag: " <> show { owner, repo, ref }
-        Git.cloneGitTag (Array.fold [ "https://github.com/", owner, "/", repo ]) ref tmpDir
-        log $ "Getting published time..."
+      LegacyPursPublish -> do
+        Log.debug $ "Cloning repo at tag: " <> show { owner, repo, ref }
+        let address = Array.fold [ "https://github.com/", owner, "/", repo ]
+        liftAff (try (Git.cloneGitTag address ref tmpDir)) >>= case _ of
+          Left error -> Notify.die $ "Could not clone " <> address <> " at ref " <> ref <> " due to a git error: " <> Aff.message error
+          Right _ -> pure unit
```

This fixes the issue with the registry silently failing when you try to publish a package with a tag that doesn't exist, by the way (see https://github.com/purescript/registry/issues/91)